src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -1053,6 +1152,67 @@ double VideoDecoder::getPtsSecondsForFrame(
   return ptsToSeconds(stream.allFrames[frameIndex].pts, stream.timeBase);
 }
 
+void VideoDecoder::convertFrameToBufferUsingSwsScale(
+    RawDecodedOutput& rawOutput) {
+  AVFrame* frame = rawOutput.frame.get();
+  int streamIndex = rawOutput.streamIndex;
+  enum AVPixelFormat frameFormat =
+      static_cast<enum AVPixelFormat>(frame->format);
+  StreamInfo& activeStream = streams_[streamIndex];
+  int outputWidth = activeStream.options.width.value_or(frame->width);
+  int outputHeight = activeStream.options.height.value_or(frame->height);
+  if (activeStream.swsContext.get() == nullptr) {
+    SwsContext* swsContext = sws_getContext(
+        frame->width,
+        frame->height,
+        frameFormat,
+        outputWidth,
+        outputHeight,
+        AV_PIX_FMT_RGB24,
+        SWS_BILINEAR,
+        nullptr,
+        nullptr,
+        nullptr);
+    const int *invTable, *table;
```

Comment:
As a rule in C++, we should declare one variable per line, and it should be pronounced `const int* invTable`. I'd also ideally like to see us explicitly initialize everything, even if it's going to be set in a function we're caling.